cmd/gctsDeployCommit.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"net/http/cookiejar"
+
+	gabs "github.com/Jeffail/gabs/v2"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func gctsDeployCommit(config gctsDeployCommitOptions, telemetryData *telemetry.CustomData) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+	httpClient := &piperhttp.Client{}
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := deployCommit(&config, telemetryData, &c, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func deployCommit(config *gctsDeployCommitOptions, telemetryData *telemetry.CustomData, command execRunner, httpClient piperhttp.Sender) error {
+
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return errors.Wrap(cookieErr, "deploy commit failed")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	url := config.Host +
+		"/sap/bc/cts_abapvcs/repository/" + config.Repository +
+		"/pullByCommit?sap-client=" + config.Client
+	if config.Commit != "" {
+		url = url + "&request=" + config.Commit
+	}
+
+	resp, httpErr := httpClient.SendRequest("GET", url, nil, nil, nil)
+
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if resp == nil || httpErr != nil {
+		return errors.Errorf("deploy commit failed: %v", httpErr)
+	}
+
+	bodyText, readErr := ioutil.ReadAll(resp.Body)
+
+	if readErr != nil {
+		return errors.Wrapf(readErr, "deploying commit failed")
```

Comment:
Isn't the error message misleading / not precise? We can judge based on the response code that something went wrong wether or not a "deployment" succeeded. But here - in case we are not able to read the response we might have other issues as well. Seems to be wrong to conclude from not being able to read the response body that a deployment failed.

Reply:
True. Since reading the response body is only "necessary" for the final success message, do you suggest simply issuing a warning here?

Reply:
We are not able to provide a corresponding message in case of a success below in case we are not able to read the response here. I guess `response.Path(./.)` will fail in this case. Hence it is OK from my point of view to raise an error here in case we can't read the response. But would be good if the message would be more precise.